File: c++/advanced/3a_var_size_min_subarray_sum_gt_target.cpp
//Minimum Size Subarray  with Sum >= target

int minSubArrayLen(int target, vector<int>& nums) {
        int l=0, sum=0, ans=INT_MAX;
        for(int r=0; r<nums.size(); r++){
            sum+= nums[r];
            while(sum-nums[l] >= target){
                sum-= nums[l];
                l++;
            }
            if(sum>= target) ans = min(ans, r-l+1);
        }
        return ans == INT_MAX ? 0 : ans;
}

File: js/advanced/3a_var_size_min_subarray_sum_gt_target.js
//Minimum Size Subarray  with Sum >= target

var minSubArrayLen = function(target, nums) {
    let l = 0, ans = Number.MAX_SAFE_INTEGER, sum = 0
    for(let r=0; r<nums.length; r++){
        sum+= nums[r]
        while(sum-nums[l] >= target){
            sum-= nums[l]
            l++
        }
        if(sum >= target) ans = Math.min(ans, r-l+1)
    }
    return ans == Number.MAX_SAFE_INTEGER? 0 : ans
};